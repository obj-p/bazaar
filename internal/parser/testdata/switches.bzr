package switches

func Switches() {
    switch expression {
    case expr1:
    case value1:
        2 + 2
        foo()
    default:
        1 + 1
    }
}
