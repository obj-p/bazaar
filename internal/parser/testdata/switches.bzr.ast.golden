&parser.Bazaar{
  Package: &parser.PackageDecl{
    Domain: []string{
      "switches",
    },
  },
  Decls: []*parser.TopLevelDecl{
    {
      Template: &parser.TemplateDecl{
        Name: "Switches",
        Block: []*parser.Stmt{
          {
            Switch: &parser.SwitchStmt{
              Expr: &parser.Expr{
                Unary: &parser.UnaryExpr{
                  Primary: &parser.PrimaryExpr{
                    Reference: &parser.ReferenceExpr{
                      Name: &"expression",
                    },
                  },
                },
              },
              Body: &parser.SwitchBody{
                Cases: []*parser.SwitchCase{
                  {
                    Expr: &parser.Expr{
                      Unary: &parser.UnaryExpr{
                        Primary: &parser.PrimaryExpr{
                          Reference: &parser.ReferenceExpr{
                            Name: &"expr1",
                          },
                        },
                      },
                    },
                  },
                  {
                    Expr: &parser.Expr{
                      Unary: &parser.UnaryExpr{
                        Primary: &parser.PrimaryExpr{
                          Reference: &parser.ReferenceExpr{
                            Name: &"value1",
                          },
                        },
                      },
                    },
                    Block: []*parser.Stmt{
                      {
                        Expr: &parser.Expr{
                          Binary: &parser.BinaryExpr{
                            Left: &parser.Expr{
                              Unary: &parser.UnaryExpr{
                                Primary: &parser.PrimaryExpr{
                                  Literal: &parser.Literal{
                                    Number: &"2",
                                  },
                                },
                              },
                            },
                            Op: token.Op(token.OpAdd),
                            Right: &parser.Expr{
                              Unary: &parser.UnaryExpr{
                                Primary: &parser.PrimaryExpr{
                                  Literal: &parser.Literal{
                                    Number: &"2",
                                  },
                                },
                              },
                            },
                          },
                        },
                      },
                      {
                        Call: &parser.CallStmt{
                          Name: &"foo",
                        },
                      },
                    },
                  },
                },
                Default: []*parser.Stmt{
                  {
                    Expr: &parser.Expr{
                      Binary: &parser.BinaryExpr{
                        Left: &parser.Expr{
                          Unary: &parser.UnaryExpr{
                            Primary: &parser.PrimaryExpr{
                              Literal: &parser.Literal{
                                Number: &"1",
                              },
                            },
                          },
                        },
                        Op: token.Op(token.OpAdd),
                        Right: &parser.Expr{
                          Unary: &parser.UnaryExpr{
                            Primary: &parser.PrimaryExpr{
                              Literal: &parser.Literal{
                                Number: &"1",
                              },
                            },
                          },
                        },
                      },
                    },
                  },
                },
              },
            },
          },
        },
      },
    },
  },
}
